### ISSUE TITLE
Google Speech API Won't Accept Large Files

### ISSUE THREAD

**Issue by brianarpie** [GoogleCloudPlatform/google-cloud-ruby#2210]
Hi, I'm receiving a server error when trying to process large audio files. The audio files are originally audio/m4a @ 32kHz and per the recommendations of the documentation am converting/compressing them to audio/amr_wb @ 16kHz. These files are well below the 180 minutes of audio limit yet I'm still receiving a server error when processing them.

`GaxError Exception occurred in retry method that was not classified as transient, caused by 8:Received message larger than max (5371623 vs. 4194304)`

I'm using version `V1p1beta` and the method `long_running_recognize` to transcribe these audio files. My files are hosted on Google Cloud Storage and I'm providing the uri in my api call.

How can I send large audio files to the API without the server enforcing a size restriction ? It seems wrong to recommend using FLAC or WAV and having a 180 minutes of length in audio limit if the server can't even handle my hour long audio file that has been encoded to AMR_WB.

Thanks!

**Comment by brianarpie** [comment]
@quartzmo, thanks for the reply. I updated my question (perhaps a little too late) and noted that I'm already storing the files in GCS and am supplying the URI to the API. I am only having issues for files that are over a certain size.

Not that I'm sure it's relevant to my issue, I forked this library and added a few things:
* ability to pass the param enable_speaker_diarization to the api (and include speaker_tag in the response)
* ability to check the status/response of a long running operation by the operation's name

https://github.com/GoogleCloudPlatform/google-cloud-ruby/compare/master...aloeai:speaker-diarization

This error is included in the response of the long running operation when I check on it (only for large files). Any other ideas?
